blackduck/Authentication.py

```diff
@@ -78,9 +162,9 @@ def authenticate(self):
                 )
 
             content = response.json()
-            self.csrf_token = response.headers.get('X-CSRF-TOKEN')
-            self.auth_token = content.get('bearerToken')
-            self.valid_until = datetime.utcnow() + timedelta(milliseconds=int(content.get('expiresInMilliseconds', 0)))
+            self.cookie = response.headers.get('Set-Cookie', 'COOKIE_UNSET')
+            self.auth_token = self.cookie[cookie.index('=')+1:cookie.index(';')]
+            self.valid_until = datetime.utcnow() + timedelta(milliseconds=int(content.get('expiresInMilliseconds', 900000)))
```

Comment:
From my testing, the bearer token expires after 2 hours for username/password. Also authentication must be attempted on first call. I'll test auto-renewal this week.